### PR TITLE
chore: expose internal api for extending injected script

### PR DESCRIPTION
**Pull request by pavelfeldman**

### CODE REVIEW

File: src/debug/debugController.ts [hunk 1]
```diff
@@ -22,16 +22,10 @@ import { Progress } from '../server/progress';
...
-
 export class DebugController implements InstrumentingAgent {
   private async ensureInstalledInFrame(frame: frames.Frame) {
     try {
-      const mainContext = await frame._mainContext();
-      if ((mainContext as any)[debugScriptSymbol])
-        return;
-      (mainContext as any)[debugScriptSymbol] = true;
-      await mainContext.extendInjectedScript(debugScriptSource.source);
+      await frame.extendInjectedScript(debugScriptSource.source);
```
Comment (dgozman) [inline src/debug/debugController.ts:28]: This is called on every navigation, including same-document. That means we can extend the same injected script twice. Is that ok?

File: src/client/frame.ts [hunk 2]
```diff
@@ -410,6 +410,11 @@ export class Frame extends ChannelOwner<channels.FrameChannel, channels.FrameIni
       return (await this._channel.title()).value;
     });
   }
+
+  async _extendInjectedScript<Arg>(source: string, arg?: Arg): Promise<JSHandle> {
```
Comment (dgozman) [inline src/client/frame.ts:414]: Do you plan to call this one?

**Review by dgozman — APPROVED** [review]

File: src/client/frame.ts (hunk 2 above)
Comment (pavelfeldman) [reply src/client/frame.ts:414]: Yes

File: src/debug/debugController.ts (hunk 1 above)
Comment (pavelfeldman) [reply src/debug/debugController.ts:28]: Clients would need to take care of it for now...

**Pull request merged**